### PR TITLE
:book: Add call out for area labels when making a PR

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
We would like users to add labels to their PRs. This will help the release team know the areas of the code base update for better release notes. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Ref #8051

#### What type of PR is this?
/area documentation

### CODE REVIEW

**Comment by joekr** [comment]
I'm not sure who to work with regarding adding labels, but we probably want to have:

- documentation (or docs)
- ClusterClass
- KCP
- we have testing but what about `e2e-testing` or `test/e2e`
- CI

@oscr might have more label suggestions as well.

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

It's relatively easy to add and remove area labels, so it's fine to take an iterative approach there. I think, if this is the agreed upon approach, we could also add a /needs-area label to all PRs that don't have one, which gives  a hint to reviewers to add the label.

**Comment by killianmuldoon** [comment]
@joekr Could you open an issue to discuss cleaning up the existing area labels and adding new useful ones?

/retitle :book: Add call out for area labels when making a PR

**Comment by sbueringer** [comment]
+1 to figure out which area labels we want via an issue. I can give some input there

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,20 @@
-<!-- please add an icon to the title of this PR (see https://sigs.k8s.io/cluster-api/CONTRIBUTING.md#contributing-a-patch), and delete this line and similar ones -->
-<!-- the icon will be either ⚠️ (:warning:, major or breaking changes), ✨ (:sparkles:, feature additions), 🐛 (:bug:, patch and bugfixes), 📖 (:book:, documentation or proposals), or 🌱 (:seedling:, minor or other) -->
+<!-- Thanks for sending a pull request! Here are some tips for you:
+
+    1. If this is your first time, please read our contributor guidelines: https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#contributing-a-patch and developer guide https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/guide.md
+
+    2. Please label this pull request according to what area(s) you are addressing. For reference on on PR/issue labels: https://github.com/kubernetes-sigs/cluster-api/labels?q=area+
```
Comment (fabriziopandini) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: should this go down into What type of PR is this?

Comment (nprokopic) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: +1 here, number 2 above sounds like it should go into "What type of PR is this?"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+-->
 
 **What this PR does / why we need it**:
 
 **Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
 Fixes #
+
+#### What type of PR is this?
+<!--
+Area example:
```
Comment (chrischdi) [inline .github/PULL_REQUEST_TEMPLATE.md:18]:
```suggestion
Example for adding an area label (see https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of available labels):
```

Comment (joekr) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: thanks for the call out. I forgot about this PR 😵‍💫

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,7 +1,21 @@
-<!-- please add an icon to the title of this PR (see https://sigs.k8s.io/cluster-api/CONTRIBUTING.md#contributing-a-patch), and delete this line and similar ones -->
-<!-- the icon will be either ⚠️ (:warning:, major or breaking changes), ✨ (:sparkles:, feature additions), 🐛 (:bug:, patch and bugfixes), 📖 (:book:, documentation or proposals), or 🌱 (:seedling:, minor or other) -->
+<!-- Thanks for sending a pull request! Here are some tips for you:
+
```
Comment (sbueringer) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,7 +1,21 @@
...
+    2. Please add an icon to the title of this PR (see https://sigs.k8s.io/cluster-api/CONTRIBUTING.md#contributing-a-patch), and delete this line and similar ones
+    the icon will be either ⚠️ (:warning:, major or breaking changes), ✨ (:sparkles:, feature additions), 🐛 (:bug:, patch and bugfixes), 📖 (:book:, documentation or proposals), or 🌱 (:seedling:, minor or other) 
+-->
 
 **What this PR does / why we need it**:
 
 **Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
 Fixes #
+
+**What type of PR is this?**
```
Comment (sbueringer) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Q: Do we need this change to the template or is it enough that we introduce the but which bugs people until they add an area label?

If we don't need it, I would prefer to keep the PR template as short as possible. At some point folks are just stop reading it. (also we have to dig through it ourselves every time we open a PR)

Comment (joekr) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: If we can have the bot do it ... then that would be amazing. Not sure how to go about doing that though 😊

Comment (sbueringer) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: There is a PR for it open: https://github.com/kubernetes/test-infra/pull/30314

Comment (chrischdi) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I think that here helps to remember setting the area (if possible) already when creating the PR.

I was part of the release team and I did know that missing area labels are an issue. But I did always forget to add them directly when I created the PR.

From happy flow its better to already set it at the beginning as PR owner and not get bugged for forgetting it all the time :D 

TLDR: I'm ok with not having this PR but I'd prefer having this information in the template.

If its too noisy, maybe we could consider making it completely a comment.

Comment (sbueringer) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I think having it completely as a comment would be an improvement which makes it more lightweigt

Comment (sbueringer) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: This would roughly mean just drop l.13-l.15

Comment (furkatgofurov7) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: +1 to @chrischdi comment generally. 

I would also personally prefer having it in PR template and reminded by prow bot additionally. That would reduce bugging PR author/reviewer hopefully and should serve as a first layer filter before prow bot starts pinging people with "you need to set" label comment.

Comment (joekr) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: I'll drop l.14-l.15

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (joekr) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: Line 14 has this called out instead.

**Comment by sbueringer** [comment]
/lgtm
/approve

/hold cancel

Thx!
